src/services/YotiSessionCompletionProcessor.ts

```diff
@@ -398,4 +411,25 @@ export class YotiSessionCompletionProcessor {
 		  });
 	  }
   }
+
+  async sendErrorMessageToIPVCore(f2fSession: ISessionItem, errorMessage: string) : Promise<any> {
+  	this.logger.error(`VC generation failed : ${errorMessage}`, {		
+  		messageCode: MessageCodes.ERROR_GENERATING_VC,
+  	});
+  	try {
+  		await this.f2fService.sendToIPVCore({
+  			sub: f2fSession.subject,
+  			state: f2fSession.state,
+  			error: "access_denied",
+  			error_description: `VC generation failed : ${errorMessage}`,
+  		});
+  	} catch (error) {
+  		this.logger.error({ message: "Failed to send VC to IPV Core Queue" }, {
+  			error,
+  			messageCode: MessageCodes.FAILED_SENDING_VC,
+  		});		
+  		throw new AppError(HttpCodesEnum.SERVER_ERROR, "Failed to send to IPV Core", { shouldThrow: true });
```

Comment:
This is interesting because it means that if the write to IPV core fails we want to throw that error rather than the error that we encountered when we created the VC - which means this takes priority over that.

Do you think that's correct? Or should we just log the error and then continue to throw whatever error triggered this sendErrorMessageToIPVCore in the first place?

Reply:
Yeah, makes sense. lets not throw the error here and just log it.